docs/query-metadata-style-guide.md

```diff
@@ -179,8 +179,13 @@ The select clause of each alert query defines the alert message that is displaye
 * The message should factually describe the problem that is being highlighted–it should not contain recommendations about how to fix the problem or value judgements.
 * Program element references should be in 'single quotes' to distinguish them from ordinary words. Quotes are not needed around substitutions (`$@`).
 * Avoid constant alert message strings and include some context, if possible. For example, `The class 'Foo' is duplicated as 'Bar'.` is preferable to `This class is duplicated here.`
+* If a reference to the current location can't be avoided use "this location" instead of "here". For example, `Bad thing at this location.` is preferable to `Bad thing here.`. 
```

Comment:
It's not obvious to me why 'this location' would be better. I don't mind the rule, because consistent writing is a good thing — but can you add a half sentence to explain the reasoning?

Reply:
I remember someone suggesting it to avoid [the "click here" anti-pattern](https://www.lamar.edu/web-communication/resources/avoid-using-click-here.html), but now I can't find that review comment.   

I've inserted a short sentence referencing the "click here" anti-pattern in the end.

Reply:
I'm wondering a bit if we should keep this line.   

@aibaars suggested in a Slack comment that the "this location" phrasing got around the general ["click-here" anti-pattern](https://www.w3.org/QA/Tips/noClickHere).  
But as @smowton points out, it might not always make the text clearer: https://github.com/github/codeql/pull/10413#issuecomment-1246527443 

Maybe the guide should be that a link-text shouldn't be "here"?

Reply:
In most cases could you just linkify the immediately preceding text, so something like

`This sensitive data is written to a logfile unescaped /here/` -> `This sensitive data is /written to a logfile unescaped/`?

Reply:
Then you can keep `here` when it refers to "[some statement, expression, argument or result on] the line of this alert" and replace it when it's referring to a second thing involved in the alert.

Reply:
I've tried to revise the text a bit, and moved the the parts about link-texts into a separate sub-section.   

>  In most cases could you just linkify the immediately preceding text.  

Yes, and the guide now includes that 👍  

------ 

I still have the guidance about "this location" when you have an alert message without links, for when you can't get around mentioning the current location in the message.

Reply:
I'm happy with the message as it is now 👍